tracing-subscriber/src/registry/sharded.rs

```diff
@@ -0,0 +1,230 @@
+use sharded_slab::{Guard, Slab};
+
+use crate::{
+    fmt::span::SpanStack,
+    registry::{LookupSpan, SpanData},
+    sync::RwLock,
+};
+use serde_json::{json, Value};
+use std::{
+    cell::RefCell,
+    collections::{HashMap, HashSet},
+    convert::TryInto,
+    fmt, io,
+    marker::PhantomData,
+    sync::{Arc, Mutex},
+};
+use tracing_core::{
+    field::{FieldSet, Visit},
+    span::{self, Id},
+    Event, Field, Interest, Metadata, Subscriber,
+};
+
+pub struct Registry {
+    spans: Arc<Slab<BigSpan>>,
+    // TODO(david): replace this with the span stack from `fmt` (this is wrong)
+    local_spans: RwLock<SpanStack>,
```

Comment:
I don't think this field is necessary at all, since the span stack lives in a thread local?

Reply:
Yep, I removed that one!